test/acs-engine-test/config.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+)
+
+type Deployment struct {
+	ClusterDefinition string `json:"cluster_definition"`
+	Location          string `json:"location"`
+}
+
+type TestConfig struct {
+	Deployments []Deployment `json:"deployments"`
+}
+
+func (c *TestConfig) Read(data []byte) error {
+	return json.Unmarshal(data, c)
+}
+
+func (c *TestConfig) Validate() error {
+	for _, d := range c.Deployments {
+		if d.ClusterDefinition == "" {
+			return errors.New("Cluster definition is not set")
+		}
+		if d.Location == "" {
+			return errors.New("Location is not set")
+		}
+	}
+	return nil
+}
+
+func getTestConfig(fname string) (*TestConfig, error) {
+	data, err := ioutil.ReadFile(fname)
+	if err != nil {
+		return nil, err
+	}
+	config := new(TestConfig)
```

Comment:
as I was reading the some go material, it's suggested that it's idiomatic to use var config TestConfig to initialize the variable with default values. or config := TestConfig{}.